staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -547,6 +553,17 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 		}
 	}
 
+	if feature.DefaultFeatureGate.Enabled(features.RequestManagement) {
+		requestManagementHookName := "generic-apiserver-request-management"
+		err := s.AddPostStartHook(requestManagementHookName, func(context PostStartHookContext) error {
+			return c.RequestManagement.Run(context.StopCh)
```

Comment:
This code has access to the main apiservers through `c.SharedInformerFactory`, but this is later in the startup process than we would like to start reading config.

Reply:
it's designed i think. here we applies a `option-->config-->completed-config-->new` flow and it's commonly used in kubernetes world

Reply:
Right.  We did not put something in options, so the story starts with config for our feature.  I followed the standard pattern after config in #80956.

I realized my first comment here is wrong.  I was thinking of reading config objects while constructing the handler chain, but have since realized that this does not make sense.  We have to initialize the filter's data structure to a useful default (as outlined in the KEP) and let the regular config object handling update that.